op-gateway/src/OPProofService.ts

```diff
@@ -5,72 +5,100 @@ import {
-];
+// OPOutputLookup contract is deployed with deterministic deployment
+// As a result, OPOutputLookup is always deployed to the same address
+// See OPOutputLookup.sol on op-verifier/contracts/OPOutputLookup.sol
+const OP_OUTPUT_LOOKUP = "0xdBe7002f0157cAf07D9636965C26C9595e54FA65"
 
 const L2_TO_L1_MESSAGE_PASSER_ADDRESS =
   '0x4200000000000000000000000000000000000016';
 
+const OPTIMISM_PORTAL_ABI = [
+  'function l2Oracle() external view returns (address)',
+  'function disputeGameFactory() external view returns (address)',
+  'function respectedGameType() external view returns (uint32)',
+];
+
 /**
  * The proofService class can be used to calculate proofs for a given target and slot on the Optimism Bedrock network.
  * It's also capable of proofing long types such as mappings or string by using all included slots in the proof.
  *
  */
 export class OPProofService implements IProofService<OPProvableBlock> {
-  private readonly l2OutputOracle: Contract;
+  private readonly l1Provider: JsonRpcProvider;
   private readonly l2Provider: JsonRpcProvider;
   private readonly helper: EVMProofHelper;
-  private readonly delay: number;
+  private readonly minAge: number;
+
+  private readonly optimismPortal: Contract;
+  private opOutputLookup: Contract;
 
   constructor(
     l1Provider: JsonRpcProvider,
     l2Provider: JsonRpcProvider,
-    l2OutputOracleAddress: string,
-    delay: number
+    optimismPortalAddress: string,
+    minAge: number
   ) {
+    this.l1Provider = l1Provider;
     this.l2Provider = l2Provider;
     this.helper = new EVMProofHelper(l2Provider);
-    this.delay = delay;
-    this.l2OutputOracle = new Contract(
-      l2OutputOracleAddress,
-      L2_OUTPUT_ORACLE_ABI,
-      l1Provider
+    this.minAge = minAge;
+
+    this.optimismPortal = new Contract(
+      optimismPortalAddress,
+      OPTIMISM_PORTAL_ABI,
+      this.l1Provider
     );
+
+    this.opOutputLookup = new Contract(
+      OP_OUTPUT_LOOKUP,
+      OPOutputLookupABI,
+      this.l1Provider,
+    )
   }
 
   /**
    * @dev Returns an object representing a block whose state can be proven on L1.
    */
   async getProvableBlock(): Promise<OPProvableBlock> {
-    /**
-     * Get the latest output from the L2Oracle. We're building the proof with this batch
-     * We go a few batches backwards to avoid errors like delays between nodes
-     *
-     */
-    const l2OutputIndex =
-      Number(await this.l2OutputOracle.latestOutputIndex()) - this.delay;
-
-    /**
-     *    struct OutputProposal {
-     *       bytes32 outputRoot;
-     *       uint128 timestamp;
-     *       uint128 l2BlockNumber;
-     *      }
-     */
-    const outputProposal = await this.l2OutputOracle.getL2Output(l2OutputIndex);
+    const block = await this.opOutputLookup.getOPProvableBlock(
+      await this.optimismPortal.getAddress(),
+      this.minAge,
+      1000000000
```

Comment:
What parameter is this? Can you leave a comment on it?

Reply:
It's Unlimited maximum age to have the gateway fetch output root regardless of maximum age parameter as it will be verified on the contract anyway. Commented